benchmarks/cugraph/standalone/bulk_sampling/trainers/dgl/trainers_cugraph_dgl.py

```diff
@@ -0,0 +1,267 @@
+        device: int, default=0
+            The CUDA device to use.
+        rank: int, default=0
+            The global rank of the worker this trainer is assigned to.
+        world_size: int, default=1
+            The number of workers in the world.
+        num_epochs: int, default=1
+            The number of training epochs to run.
+        sample_dir: str, default="."
+            The directory where samples generated by the bulk sampler
+            are stored.
+        backend: str, default="torch"
+            The feature store backend to be used by the cuGraph Feature Store.
+            Defaults to "torch".  Options are "torch" and "wholegraph"
+        kwargs
+            Keyword arguments to pass to the loader
+        """
+        self.__data = None
+        self.__device = device
+        self.__rank = rank
+        self.__world_size = world_size
+        self.__num_epochs = num_epochs
+        self.__dataset = dataset
+        self.__sample_dir = sample_dir
+        self.__loader_kwargs = kwargs
+        self.__model = self.get_model(model)
+        self.__optimizer = None
+        self.__backend = backend
+
+    @property
+    def rank(self):
+        return self.__rank
+
+    @property
+    def model(self):
+        return self.__model
+
+    @property
+    def dataset(self):
+        return self.__dataset
+
+    @property
+    def optimizer(self):
+        if self.__optimizer is None:
+            self.__optimizer = torch.optim.Adam(
+                self.model.parameters(), lr=0.01, weight_decay=0.0005
+            )
+        return self.__optimizer
+
+    @property
+    def num_epochs(self) -> int:
+        return self.__num_epochs
+
+    def get_loader(self, epoch: int = 0, stage="train") -> int:
+        # TODO support online sampling
+        if stage == "train":
+            path = os.path.join(self.__sample_dir, f"epoch={epoch}", stage, "samples")
+        elif stage in ["test", "val"]:
+            path = os.path.join(self.__sample_dir, stage, "samples")
+        else:
+            raise ValueError(f"Invalid stage {stage}")
+
+        dataloader = get_dataloader(
+            input_file_paths=self.get_input_files(
+                path, epoch=epoch, stage=stage
+            ).tolist(),
+            total_num_nodes=None,
+            sparse_format="csc",
+            return_type="cugraph_dgl.nn.SparseGraph",
+        )
+        return dataloader
+
+    @property
+    def data(self):
+        import logging
+
+        logger = logging.getLogger("DGLCuGraphTrainer")
+        logger.info("getting data")
+
+        if self.__data is None:
+            # FIXME wholegraph
+            fs = FeatureStore(backend=self.__backend)
+            num_nodes_dict = {}
+
+            if self.__backend == "wholegraph":
+                from pylibwholegraph.torch.initialize import get_global_communicator
+
+                wm_comm = get_global_communicator()
+                wm_comm.barrier()
```

Comment:
Dumb question: 

The wholegraph dataset gets shared across workers here right , does that happen by default ?